src/search-manager/data/api.ts

```diff
@@ -303,6 +303,29 @@ export async function fetchSearchResults({
   };
 }
 
+/**
+ * Fetch the block types facet distribution for the search results.
+ */
+export const fetchBlockTypes = async (
+  client: MeiliSearch,
+  indexName: string,
+  extraFilter?: Filter,
+): Promise<Record<string, number>> => {
+  // Convert 'extraFilter' into an array
+  const extraFilterFormatted = forceArray(extraFilter);
+
+  const { results } = await client.multiSearch({
+    queries: [{
+      indexUid: indexName,
+      facets: ['block_type'],
+      filter: extraFilterFormatted,
+      limit: 0, // We don't need any "hits" for this - just the facetDistribution
+    }],
+  });
+
+  return results[0].facetDistribution?.block_type ?? {};
+};
```

Comment:
I thought we already had this information in the 2nd multisearch query? https://github.com/openedx/frontend-app-authoring/blob/95c17537c16cf6d3bb9ad4b3a2742dd23a409728/src/search-manager/data/api.ts#L255-L266

Reply:
When we are on the library page and open the collection sidebar, the 2nd search has the facets for all blocks in this library (not only in the collection).